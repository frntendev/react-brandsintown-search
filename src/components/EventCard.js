import React from "react";
import PropTypes from "prop-types";
import { css } from "react-emotion";

const styles = {
  event: css`
    display: flex;
    min-height: 50px;
    background: #eeeeee85;
    margin-top: 10px;
    margin-bottom: 10px;
    align-items: center;
    padding: 5px;
    box-sizing: border-box;
    transition: 0.3s all ease;
    text-align: center;
    &:hover {
      box-shadow: 0 10px 20px rgba(0, 0, 0, 0.19), 0 6px 6px rgba(0, 0, 0, 0.23);
    }
    label: event-root;
  `,
  eventText: css`
    flex: 1;
    font-size: 1.2rem;
    color: #545454;
    label: eventText;
    @media (max-width: 758px) {
      font-size: 1rem;
    }
  `,
  eventLink: css`
    background: #48c5b6;
    color: #fff;
    text-decoration: none;
    font-size: 1.2rem;
    padding: 10px;
    white-space: pre;
    transition: 0.3s all ease;
    margin-left: 5px;
    cursor: pointer;
    label: eventLink;
    @media (max-width: 758px) {
      font-size: 1rem;
    }
    &:hover {
      background: #3fa99c;
    }
  `
};
const EventCard = props => {
  return (
    <div className={styles.event}>
      <span className={styles.eventText}>
        <strong>{props.country}</strong>
      </span>
      <span className={styles.eventText}>{props.city}</span>
      <span className={styles.eventText}>{props.date}</span>
      <span className={styles.eventLink} onClick={props.onSaveEvent}>
        Save
      </span>
      <a className={styles.eventLink} href={props.url} target="_blank">
        Book
      </a>
    </div>
  );
};

export default EventCard;

EventCard.defaultProps = {
  country: "Iran",
  city: "Tehran",
  date: "1-1-2018"
};

EventCard.propTypes = {
  /**
   * The country of an event
   */
  country: PropTypes.string.isRequired,
  /**
   * The city of an event
   */
  city: PropTypes.string.isRequired,
  /**
   * The date of an event
   */
  date: PropTypes.string.isRequired,
  /**
   * The url of an event
   */
  url: PropTypes.string,
  /**
   * A callback function that saves the favorite event
   */
  onSaveEvent: PropTypes.func
};
