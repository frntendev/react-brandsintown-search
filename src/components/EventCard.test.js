import React from "react";
import { shallow, mount, configure } from "enzyme";
import EventCard from "./EventCard";
import Adapter from "enzyme-adapter-react-16";
configure({ adapter: new Adapter() });

describe("EventCard", () => {
  test("renders component itself correctly", () => {
    const renderedComponent = shallow(<EventCard />);
    expect(renderedComponent).toMatchSnapshot();
  });

  test("renders component props correctly", () => {
    const renderedComponent = mount(
      <EventCard country="Germany" city="Berlin" />
    );
    expect(renderedComponent.props().country).toEqual("Germany");
    expect(renderedComponent.props().city).toEqual("Berlin");
  });
});
