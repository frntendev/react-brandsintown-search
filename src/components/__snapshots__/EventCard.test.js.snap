// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventCard renders component itself correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <EventCard
    city="Tehran"
    country="Iran"
    date="1-1-2018"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <span
          className="css-q8zcde-eventText"
        >
          <strong>
            Iran
          </strong>
        </span>,
        <span
          className="css-q8zcde-eventText"
        >
          Tehran
        </span>,
        <span
          className="css-q8zcde-eventText"
        >
          1-1-2018
        </span>,
        <span
          className="css-bu8z06-eventLink"
        >
          Save
        </span>,
        <a
          className="css-bu8z06-eventLink"
          target="_blank"
        >
          Book
        </a>,
      ],
      "className": "css-f67bwj-event-root",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <strong>
            Iran
          </strong>,
          "className": "css-q8zcde-eventText",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Iran",
          },
          "ref": null,
          "rendered": "Iran",
          "type": "strong",
        },
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Tehran",
          "className": "css-q8zcde-eventText",
        },
        "ref": null,
        "rendered": "Tehran",
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "1-1-2018",
          "className": "css-q8zcde-eventText",
        },
        "ref": null,
        "rendered": "1-1-2018",
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Save",
          "className": "css-bu8z06-eventLink",
          "onClick": undefined,
        },
        "ref": null,
        "rendered": "Save",
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Book",
          "className": "css-bu8z06-eventLink",
          "href": undefined,
          "target": "_blank",
        },
        "ref": null,
        "rendered": "Book",
        "type": "a",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <span
            className="css-q8zcde-eventText"
          >
            <strong>
              Iran
            </strong>
          </span>,
          <span
            className="css-q8zcde-eventText"
          >
            Tehran
          </span>,
          <span
            className="css-q8zcde-eventText"
          >
            1-1-2018
          </span>,
          <span
            className="css-bu8z06-eventLink"
          >
            Save
          </span>,
          <a
            className="css-bu8z06-eventLink"
            target="_blank"
          >
            Book
          </a>,
        ],
        "className": "css-f67bwj-event-root",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <strong>
              Iran
            </strong>,
            "className": "css-q8zcde-eventText",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Iran",
            },
            "ref": null,
            "rendered": "Iran",
            "type": "strong",
          },
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Tehran",
            "className": "css-q8zcde-eventText",
          },
          "ref": null,
          "rendered": "Tehran",
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "1-1-2018",
            "className": "css-q8zcde-eventText",
          },
          "ref": null,
          "rendered": "1-1-2018",
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Save",
            "className": "css-bu8z06-eventLink",
            "onClick": undefined,
          },
          "ref": null,
          "rendered": "Save",
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Book",
            "className": "css-bu8z06-eventLink",
            "href": undefined,
            "target": "_blank",
          },
          "ref": null,
          "rendered": "Book",
          "type": "a",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
